### ISSUE TITLE
Rule proposal: `return condition` instead of using if/else or ternary with explicit true/false

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#1219]
Not to be confused with https://eslint.org/docs/rules/no-else-return 
Complementary to https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-ternary.md

Occasionally I see Boolean functions that include multiple returns where a plain `return condition` would suffice.

## Fail

```js
function isLoggedIn() {
	if (state === 'logged-in') {
		return true;
	}

	return false;
}
```
```js
function isTruthy(value) {
	if (value) {
		return true;
	}

	return false;
}
```

```js
function isNumber(value) {
	return typeof value === 'number' ? true : false;
}
```

## Pass

```js
function isLoggedIn() {
	return state === 'logged-in';
}
```
```js
function isTruthy(value) {
	return Boolean(value);
}
```

```js
function isNumber(value) {
	return typeof value === 'number';
}
```

**Comment by fisker** [comment]
#1116 , same?

**Comment by fisker** [comment]
This won't be a new rule, we'll add this to `prefer-ternary`, we just need to detect this case. Previous related discussion https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1116

**Comment by fregante** [comment]
#1116 is not the same and there's no ternary here. Both rules are mentioned in this issue’s text as not being the same

**Comment by fisker** [comment]
AH, sorry... misunderstood.

**Comment by fregante** [comment]
> read too quick

No worries, I do the same 😃

**Comment by fisker** [comment]
But, if we do #1116, then [`no-unneeded-ternary`](https://eslint.org/docs/rules/no-unneeded-ternary) will catch this, right?

**Comment by fregante** [comment]
Good point, `prefer-ternary` would change

```js
function isLoggedIn() {
	if (state === 'logged-in') {
		return true;
	}

	return false;
}
```

to

```js
function isLoggedIn() {
	return state === 'logged-in' ? true : false;
}
```

and `no-unneeded-ternary` changes that to

```js
function isLoggedIn() {
	return state === 'logged-in';
}
```

**Comment by fregante** [comment]
Ok I'm closing this because it's technically already possible, but there's a problem:

### `prefer-ternary` doesn't apply to the result of `no-else-return`

`no-else-return` changes this

```js
function isLoggedIn() {
	if (document.isConnected) {
		return true;
	} else {
		return false;
	}
}
```

to 

```js
function isLoggedIn() {
	if (document.isConnected) {
		return true;
	}

	return false;
}
```

And then no other rule applies. `prefer-ternary` should still catch this.

If I disable `no-else-return`, the first example is changed to the desired:

```js
function isLoggedIn() {
	return Boolean(document.isConnected);
}
```

**Comment by fisker** [comment]
I'll find time to fix #1116.

**Comment by fisker** [comment]
I'll find time to fix #1116.
